packages/generator/src/v1.1/generateQueries.ts

```diff
@@ -147,9 +147,9 @@ function handleQueryDataType(
       return `{ ${properties.join(",\n")} }`;
 
     case "union":
-      return dataType.unionTypes.map(type =>
-        handleQueryDataType(type, importedObjects)
-      ).join("|");
+      return dataType.unionTypes.filter(subType =>
+        !isNullableQueryDataType(subType)
+      ).map(type => handleQueryDataType(type, importedObjects)).join("|");
```

Comment:
It might be safer to do the filter step last instead of first and just delete "null".

union(string, union(number, null)) <- something like this